hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieHadoopFsRelationFactory.scala

```diff
@@ -229,7 +229,7 @@ class HoodieMergeOnReadSnapshotHadoopFsRelationFactory(override val sqlContext:
       recordMergerImpls = recordMergerImpls,
       recordMergerStrategy = recordMergerStrategy
     )
-  val mandatoryFields: Seq[String] = mandatoryFieldsForMerging
+  val mandatoryFields: Seq[String] = if (isBootstrap) Seq.empty else partitionColumnsToRead.toSeq
```

Comment:
for now we need to make bootstrap not read from files. But we can do it for most partition columns, just not the ones in the bootstrap base path that aren't written to files